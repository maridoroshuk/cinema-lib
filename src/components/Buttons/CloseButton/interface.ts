export interface ICloseButtonProps {
  onClose: () => void;
}
