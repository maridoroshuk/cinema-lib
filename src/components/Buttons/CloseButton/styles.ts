import styled from 'styled-components';

export const ButtonWrapper = styled.div`
  position: absolute;
  top: ${({ theme }) => theme.sizes.width[10]}px;
  right: ${({ theme }) => theme.sizes.width[20]}px;
  z-index: ${({ theme }) => theme.zIndex[5000]};

  @media (min-width: ${({ theme }) => theme.screenSizes[860]}px) {
    top: ${({ theme }) => theme.sizes.width[40]}px;
    right: ${({ theme }) => theme.sizes.width[50]}px;

  }
`;

export const StyledButton = styled.button`
  width: ${({ theme }) => theme.sizes.width[30]}px;
  height: ${({ theme }) => theme.sizes.width[30]}px;
  border-radius: 50%;
  background-color: ${({ theme }) => theme.colors.neutral300Transparent};
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  position: relative;
  border: none;
  &:before,
  &:after {
    content: '';
    position: absolute;
    width: ${({ theme }) => theme.sizes.width[20]}px;
    height: ${({ theme }) => theme.sizes.height[2]}px;
    background-color: ${({ theme }) => theme.colors.neutral000};
  }
  &:before {
    transform: rotate(45deg);
  }
  &:after {
    transform: rotate(-45deg);
  }

  @media (min-width: ${({ theme }) => theme.screenSizes[860]}px) {
    width: ${({ theme }) => theme.sizes.width[50]}px;
    height: ${({ theme }) => theme.sizes.width[50]}px;
    &:before,
    &:after {
      content: '';
      position: absolute;
      width: ${({ theme }) => theme.sizes.width[30]}px;
      height: ${({ theme }) => theme.sizes.height[2]}px;
      background-color: ${({ theme }) => theme.colors.neutral000};
    }
  }
`;
