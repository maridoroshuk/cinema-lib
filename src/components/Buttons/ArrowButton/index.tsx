import React from 'react';

import ArrowImage from '@/assets/icons/arrow.png';

import { Arrow, Wrapper } from './styles';
import { IArrowButton } from './interface';

export const ArrowButton = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Arrow src={ArrowImage} alt="arrow (hide)" />
    </Wrapper>
  );
};
