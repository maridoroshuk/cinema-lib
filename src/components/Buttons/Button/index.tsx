import React from 'react';

import { ButtonStyled } from '@/components/Buttons/Button/styles';

import { IButtonProps } from './interface';

export const Button = ({
  children,
  size,
  backgroundColor,
  type,
  value,
  onClick,
  dataCy,
}: IButtonProps) => {
  return (
    <ButtonStyled
      backgroundColor={backgroundColor}
      size={size}
      type={type || 'button'}
      value={value}
      onClick={onClick}
      data-cy={dataCy}
    >
      {children}
    </ButtonStyled>
  );
};
