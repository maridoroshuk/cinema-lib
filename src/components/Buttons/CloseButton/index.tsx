import React from 'react';
import { ICloseButtonProps } from './interface';
import { ButtonWrapper, StyledButton } from './styles';

export const CloseButton = ({ onClose }: ICloseButtonProps) => {
  return (
    <ButtonWrapper>
      <StyledButton onClick={onClose} />
    </ButtonWrapper>
  );
};
