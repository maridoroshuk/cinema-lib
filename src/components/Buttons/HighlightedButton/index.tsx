import React from 'react';
import { IUnderlineButton } from './interface';
import { ButtonStyled } from './styles';

export const HighlightedButton = ({
  children,
  handleButtonClick,
  value,
}: IUnderlineButton) => {
  return (
    <ButtonStyled onClick={handleButtonClick} value={value || ''}>
      {children}
    </ButtonStyled>
  );
};
