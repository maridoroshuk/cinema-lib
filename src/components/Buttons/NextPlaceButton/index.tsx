import React from 'react';

import NextPlaceButton from '@/assets/icons/next_place_button.png';

import { Next, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const NextPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Next src={NextPlaceButton} />
    </Wrapper>
  );
};
