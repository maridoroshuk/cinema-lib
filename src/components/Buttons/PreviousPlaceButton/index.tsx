import React from 'react';

import PreviousPlaceButton from '@/assets/icons/previous_place_button.png';

import { Previous, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const PreviousPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Previous src={PreviousPlaceButton} />
    </Wrapper>
  );
};
