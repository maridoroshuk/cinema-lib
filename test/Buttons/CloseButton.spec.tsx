import React from 'react';
import { render } from '@testing-library/react';

import { CloseButton } from '@/components/Buttons/CloseButton';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('CloseButton', () => {
  it('renders', () => {
    const handleClose = jest.fn();
    const { container } = render(
      setupTestProvider(<CloseButton onClose={handleClose} />),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
