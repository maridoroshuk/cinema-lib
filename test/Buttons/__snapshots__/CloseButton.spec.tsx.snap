// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders 1`] = `
.c0 {
  position: absolute;
  top: 40px;
  right: 50px;
  z-index: 5000;
}

.c1 {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  background-color: #d9d9d933;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  position: relative;
  border: none;
}

.c1:before,
.c1:after {
  content: '';
  position: absolute;
  width: 30px;
  height: 2px;
  background-color: #FFFFFF;
}

.c1:before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c1:after {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

<div
  class="c0"
>
  <button
    class="c1"
  />
</div>
`;
